swift/Tasks/DittoManager.swift

```diff
@@ -2,41 +2,368 @@ import DittoSwift
+    /// - Using the INITIAL keyword to only insert if documents don't already exist
+    ///
+    /// The initial tasks include common todo items like:
+    /// - Buy groceries
+    /// - Clean the kitchen
+    /// - Schedule dentist appointment
+    /// - Pay bills
+    ///
+    /// - SeeAlso: https://docs.ditto.live/sdk/latest/crud/read#using-args-to-query-dynamic-values///
+    ///
+    /// - Throws: A DittoError if the insert operations fail
+    func populateTaskCollection() async throws {
+
+        let initialTasks: [TaskModel] = [
+            TaskModel(
+                _id: "50191411-4C46-4940-8B72-5F8017A04FA811",
+                title: "Buy groceries"
+            ),
+            TaskModel(
+                _id: "6DA283DA-8CFE-4526-A6FA-D385089364E811",
+                title: "Clean the kitchen"
+            ),
+            TaskModel(
+                _id: "5303DDF8-0E72-4FEB-9E82-4B007E5797F911",
+                title: "Schedule dentist appointment"
+            ),
+            TaskModel(
+                _id: "38411F1B-6B49-4346-90C3-0B16CE97E17911",
+                title: "Pay bills"
+            ),
+        ]
+
+        for task in initialTasks {
+            if let ditto {
+                // https://docs.ditto.live/sdk/latest/crud/write#inserting-documents
+                try await ditto.store.execute(
+                    query: "INSERT INTO tasks INITIAL DOCUMENTS (:task)",
+                    arguments: [
+                        "task":
+                            [
+                                "_id": task._id,
+                                "title": task.title,
+                                "done": task.done,
+                                "deleted": task.deleted,
+                            ]
+                    ]
+                )
+            }
+        }
+    }
+
+    /// Registers observers for the planets collection to handle real-time updates.
+    ///
+    /// This method sets up a live query observer that:
+    /// - Monitors the tasks collection for changes
+    /// - Updates the @Published tasks array when changes occur
+    /// - Filters out tasks NOT deleted
+    ///
+    /// - SeeAlso: https://docs.ditto.live/sdk/latest/crud/read
+    ///
+    /// - Throws: A DittoError if the observer cannot be registered
+    func registerObservers() throws {
+        if let ditto {
+
+            let observerQuery = "SELECT * FROM tasks WHERE NOT deleted"
+            storeObserver = try ditto.store.registerObserver(
+                query: observerQuery
+            ) {
+                [weak self] results in
+                Task { @MainActor in
+                    // Create new TaskModel instances and update the published property
+                    self?.tasks = results.items.compactMap {
+                        TaskModel($0.jsonString())
+                    }
+                }
+            }
+        }
+    }
```

Comment:
Multiple occurences of this issue: `ditto` is not available if it hasn't been initialized yet, or if the API is used after it's been deinitialized. This is clearly an mistake and not doing anything here is a silent fail. Make sure this is properly handled and surfaced to the UI (or at the very least log an error).

```suggestion
    func registerObservers() throws {
        if let ditto {

            let observerQuery = "SELECT * FROM tasks WHERE NOT deleted"
            storeObserver = try ditto.store.registerObserver(
                query: observerQuery
            ) {
                [weak self] results in
                Task { @MainActor in
                    // Create new TaskModel instances and update the published property
                    self?.tasks = results.items.compactMap {
                        TaskModel($0.jsonString())
                    }
                }
            }
        } else {
            // TODO: throw an appropriate error here.
        }
    }
```